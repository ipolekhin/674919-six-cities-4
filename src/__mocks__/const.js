export const TownType = {
  AMSTERDAM: `Amsterdam`,
  BRUSSELS: `Brussels`,
  COLOGNE: `Cologne`,
  DUSSELDORF: `Dusseldorf`,
  HAMBURG: `Hamburg`,
  PARIS: `Paris`,
};

export const TOWN_NAMES = [
  TownType.AMSTERDAM,
  TownType.BRUSSELS,
  TownType.COLOGNE,
  TownType.DUSSELDORF,
  TownType.HAMBURG,
  TownType.PARIS,
];

export const TownCoordinates = {
  [TownType.AMSTERDAM]: [52.38333, 4.9],
  [TownType.BRUSSELS]: [52.38333, 4.9],
  [TownType.COLOGNE]: [52.38333, 4.9],
  [TownType.DUSSELDORF]: [52.38333, 4.9],
  [TownType.HAMBURG]: [52.38333, 4.9],
  [TownType.PARIS]: [52.38333, 4.9],
};
