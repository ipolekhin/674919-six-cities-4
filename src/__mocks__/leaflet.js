const leaflet = jest.genMockFromModule(`leaflet`);

leaflet.marker = () => {
  return {addTo: () => {
    return {};
  }};
};

leaflet.tileLayer = () => {
  return {addTo: () => {
    return {};
  }};
};

export default leaflet;
