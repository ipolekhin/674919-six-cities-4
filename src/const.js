export const AppRoute = {
  FAVORITES: `/favorites`,
  OFFER: `/offer/:id`,
  ROOT: `/`,
  SIGN_IN: `/login`,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const CARD_TYPES = [
  `Apartment`,
  `Room`,
  `House`,
  `Hotel`,
];

export const errorMessageList = {
  '400': `Сервер не понимает запрос из-за неверного синтаксиса.`,
  '404': `Сервер не может найти запрашиваемый ресурс.`,
  '0': `Сервер не отвечает.`,
};

export const FIVE_STAR = [5, 4, 3, 2, 1];

export const MapProps = {
  ICON_SIZE: [27, 39],
  ICON_URL: `img/pin.svg`,
  ICON_ACTIVE_URL: `img/pin-active.svg`,
  ZOOM: 12,
};

export const OfferCardsClassesType = {
  FAVORITES_CONTAINER: `favorites`,
  MAIN_CONTAINER: `cities__place-card`,
  OFFER_CONTAINER: `near-places`,
};

export const Ratings = [`20%`, `40%`, `60%`, `80%`, `100%`];

export const STARS_PROPERTY = [`terribly`, `badly`, `not bad`, `good`, `perfect`];

export const TownType = {
  AMSTERDAM: `Amsterdam`,
  BRUSSELS: `Brussels`,
  COLOGNE: `Cologne`,
  DUSSELDORF: `Dusseldorf`,
  HAMBURG: `Hamburg`,
  PARIS: `Paris`,
};

export const TOWN_NAMES = [
  TownType.AMSTERDAM,
  TownType.BRUSSELS,
  TownType.COLOGNE,
  TownType.DUSSELDORF,
  TownType.HAMBURG,
  TownType.PARIS,
];

export const TownCoordinates = {
  [TownType.AMSTERDAM]: [52.374, 4.889],
  [TownType.BRUSSELS]: [50.850, 4.348],
  [TownType.COLOGNE]: [50.933, 6.950],
  [TownType.DUSSELDORF]: [51.221, 6.776],
  [TownType.HAMBURG]: [53.575, 10.015],
  [TownType.PARIS]: [48.853, 2.348],
};

export const SortType = {
  POPULAR: `Popular`,
  PRICE_LOW: `Price: low to high`,
  PRICE_HIGH: `Price: high to low`,
  TOP_RATED_FIRST: `Top rated first`,
};

export const SORT_NAMES = [
  SortType.POPULAR,
  SortType.PRICE_LOW,
  SortType.PRICE_HIGH,
  SortType.TOP_RATED_FIRST,
];

export const valid = {
  // MIN_LENGTH_REVIEW: 5,
  MIN_LENGTH_REVIEW: 50,
  MAX_LENGTH_REVIEW: 300,
  TEXT_WRONG: `Текст отзыва должен содержать от 50 до 300 символов.`,
};
