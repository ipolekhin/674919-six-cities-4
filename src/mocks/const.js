export const CARD_IMAGES = [
  `img/apartment-01.jpg`,
  `img/room.jpg`,
  `img/apartment-02.jpg`,
  `img/apartment-03.jpg`,
];

export const CARD_NAMES = [
  `Beautiful & luxurious apartment at great location`,
  `Wood and stone place`,
  `Canal View Prinsengracht`,
  `Nice, cozy, warm big bed apartment`,
  `Stone and Wood place`,
];

export const INSIDE_ITEMS = [
  `Wi-Fi`,
  `Washing machine`,
  `Towels`,
  `Heating`,
  `Coffee machine`,
  `Baby seat`,
  `Kitchen`,
  `Dishwasher`,
  `Cabel TV`,
  `Fridge`,
];
