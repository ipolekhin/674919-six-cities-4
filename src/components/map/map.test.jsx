import React from "react";
import renderer from "react-test-renderer";
import Map from "./map.jsx";
import {TownCoordinates, TownType} from "../../__mocks__/const";

const placeCards = [
  {
    adults: 3,
    bedrooms: 2,
    cardName: `Wood and stone place`,
    cardRating: 4.6,
    cardRatingStars: `92%`,
    cardType: `Room`,
    coordinatesItem: [52.3909553943508, 4.85309666406198],
    description: `Text text....`,
    id: `1`,
    image: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-02.jpg`],
    insideItems: [`wi-fi`, `bathroom`],
    premiumPlace: true,
    price: 100,
    user: {
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`,
      pro: true,
    },
  },
  {
    adults: 2,
    bedrooms: 1,
    cardName: `Wood and stone place`,
    cardRating: 3.6,
    cardRatingStars: `92%`,
    cardType: `Room`,
    coordinatesItem: [52.369553943508, 4.85309666406198],
    description: `Text text....`,
    id: `2`,
    image: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-02.jpg`],
    insideItems: [`bathroom`],
    premiumPlace: true,
    price: 120,
    user: {
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`,
      pro: true,
    },
  },
];

const cityCoordinate = TownCoordinates[TownType.AMSTERDAM];

it(`Render Map`, () => {
  const tree = renderer
    .create(<Map
      placeCards = {placeCards}
      cityCoordinate = {cityCoordinate}
    />, {
      createNodeMock: () => {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
