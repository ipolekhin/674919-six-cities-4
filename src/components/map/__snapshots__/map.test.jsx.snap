// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Map 1`] = `
<div
  className="cities__right-section"
>
  <section
    className="cities__map map"
  />
</div>
`;
